proxy-socket-udp/src/test/java/net/airvantage/proxysocket/udp/ProxyDatagramSocketIPMappingTest.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * MIT License
+ * Copyright (c) 2025 Semtech
+ */
+package net.airvantage.proxysocket.udp;
+
+import net.airvantage.proxysocket.core.ProxyAddressCache;
+import net.airvantage.proxysocket.core.ProxyProtocolMetricsListener;
+import net.airvantage.proxysocket.core.v2.ProxyHeader;
+import net.airvantage.proxysocket.core.v2.ProxyProtocolV2Encoder;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Unit tests for ProxyDatagramSocket IP address mapping and cache behavior.
+ */
+class ProxyDatagramSocketIPMappingTest {
+
+    private ProxyDatagramSocket socket;
+    private ProxyAddressCache mockCache;
+    private ProxyProtocolMetricsListener mockMetrics;
+    private int localPort;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        mockCache = mock(ProxyAddressCache.class);
+        mockMetrics = mock(ProxyProtocolMetricsListener.class);
+
+        socket = new ProxyDatagramSocket(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0))
+                .setCache(mockCache)
+                .setMetrics(mockMetrics)
+                .setTrustedProxy(addr -> true); // Trust all for these tests
```

Comment:
reading the code I understand that not providing TrustedProxy is also accepted all